### PR TITLE
runtime: Allow to overwrite DESTDIR

### LINKED ISSUE #1182: runtime: Allow to easily overwrite DESTDIR
**Which feature do you think can be improved?**

If I want to install the runtime and configurations in a directory other than "/" then I need to:
```
cd src/runtime
make DESTDIR=/path/to/dir
make install
```

**How can it be improved?**

In other to ease the overwrite of DESTDIR in the CI script it could be changed the `src/runtime/Makefile` to use that variable if exported. So one can do:
```
cd src/runtime
export DESTDIR=/path/to/dir
make
make install
```

### PR DESCRIPTION
**Pull request by wainersm**
On runtime/Makefile the value of DESTDIR is set to "/", unless one
pass that variable as an argument to `make`. This change will
allow its overwrite if DESTDIR is exported in the environment as
well.

Fixes #1182

Tested as:
```
# cd src/runtime/
# mkdir -p /tmp/foo
# export DESTDIR=/tmp/foo
# make
     GENERATE data/kata-collect-data.sh
     GENERATE /root/go/src/github.com/kata-containers/kata-containers/src/runtime/cli/config-generated.go
     GENERATE pkg/katautils/config-settings.go
kata-runtime - version 2.0.0-rc0 (commit 4fbc5dbd51249795c0ca052a234ffbf671fb5cd9)

• No GO command or GOPATH not set:
	Can only install prebuilt binaries

• hypervisors:
	Default: qemu
	Known: acrn cloud-hypervisor firecracker qemu
	Available for this architecture: acrn cloud-hypervisor firecracker qemu

• Features:
	SELinux (FEATURE_SELINUX): no

• Summary:

	destination install path (DESTDIR) : /tmp/foo
	binary installation path (BINDIR) : /usr/local/bin
	binaries to install :
<snip>
# make install
# find /tmp/foo
/tmp/foo
/tmp/foo/usr
/tmp/foo/usr/libexec
/tmp/foo/usr/libexec/kata-containers
/tmp/foo/usr/libexec/kata-containers/kata-netmon
/tmp/foo/usr/share
/tmp/foo/usr/share/defaults
/tmp/foo/usr/share/defaults/kata-containers
/tmp/foo/usr/share/defaults/kata-containers/configuration.toml
/tmp/foo/usr/share/defaults/kata-containers/configuration-acrn.toml
/tmp/foo/usr/share/defaults/kata-containers/configuration-fc.toml
/tmp/foo/usr/share/defaults/kata-containers/configuration-clh.toml
/tmp/foo/usr/share/defaults/kata-containers/configuration-qemu.toml
/tmp/foo/usr/share/bash-completion
/tmp/foo/usr/share/bash-completion/completions
/tmp/foo/usr/share/bash-completion/completions/kata-runtime
/tmp/foo/usr/local
/tmp/foo/usr/local/bin
/tmp/foo/usr/local/bin/kata-monitor
/tmp/foo/usr/local/bin/containerd-shim-kata-v2
/tmp/foo/usr/local/bin/kata-runtime
/tmp/foo/usr/local/bin/kata-collect-data.sh
```

### CODE REVIEW

File: src/runtime/Makefile [hunk 1]
```diff
@@ -68,7 +68,9 @@ NETMON_TARGET = $(PROJECT_TYPE)-netmon
 NETMON_TARGET_OUTPUT = $(CURDIR)/$(NETMON_TARGET)
 BINLIBEXECLIST += $(NETMON_TARGET)
 
+ifeq ($(DESTDIR),)
```
Comment (devimc) [inline src/runtime/Makefile:71]: why don't use `DESTDIR ?= /` ?

**Review by devimc — COMMENTED** [review]
thanks @wainersm - I have a question

File: src/runtime/Makefile (hunk 1 above)
Comment (wainersm) [reply src/runtime/Makefile]: @devimc you are right. The v2 of this PR I changed it to use `?=`. Thanks for the review!

**Review by devimc — APPROVED** [review]
thanks @wainersm

**Review by fidencio — APPROVED** [review]
lgtm, thanks @wainersm

**Review by GabyCT — APPROVED** [review]

**Comment by egernst** [comment]
Once CI is working again, this LGTM!

**Review by c3d — APPROVED** [review]
lgtm

**Pull request merged**